ledger/src/blockstore_meta.rs

```diff
@@ -214,6 +232,24 @@ impl SlotMeta {
         Some(self.consumed) == self.last_index.map(|ix| ix + 1)
     }
 
+    pub fn is_connected(&self) -> bool {
+        self.connected_flags.contains(ConnectedFlags::CONNECTED)
+    }
+
+    pub fn is_parent_connected(&self) -> bool {
+        self.connected_flags
+            .contains(ConnectedFlags::PARENT_CONNECTED)
+    }
+
+    pub fn set_connected(&mut self, value: bool) {
+        self.connected_flags.set(ConnectedFlags::CONNECTED, value);
+    }
+
+    pub fn set_parent_connected(&mut self, value: bool) {
+        self.connected_flags
```

Comment:
I think we need some assertion or check here to make sure everything works correctly.

* If PARENT_CONNECTED can only be changed from false to true, or true to true, then we need to add some assertion here.  (i.e., only allow PARENT_CONNECTED to be updated when PARENT_CONNECTED is false).
* If the above statement is false, then we need to make sure when we set PARENT_CONNECTED from true to false, then we also set the CONNECTED flag to false.

Reply:
I think some assertions could make sense / help illustrate the mechanism. Let me look at something with how we purge slots and come back to this comment

Reply:
Current usage never has us doing `true ==> false` transitions on either of these bits. In purges, if we purge slot `S`, we do not update `S`'s children to remove `S` as their parent slot. I think the idea here is that we're constantly purging the oldest data when we're full, so doesn't make sense to update metas, mark orphans.

Reply:
I did the following:
- Added an assertion that parent is connected in order to set connected
- I removed the `value` parameter since our transitions are only `false` ==> `true`
- I made snapshot slots get both bits set in the new function